grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1600 @@
+                    dualAxis={dualAxis}
+                    showTickMarks={true}
+                    horizontalAxisLabelsOnTop={true}
+                />
+                <HorizontalCategoricalColorLegend manager={this} />
+                {this.renderBars()}
+            </g>
+        )
+    }
+
+    private renderBars(): JSX.Element[] {
+        const normalElements: JSX.Element[] = []
+        const highlightedElements: JSX.Element[] = [] // highlighted elements have a thicker stroke and should be drawn last to overlap others
+        const {
+            dualAxis,
+            x0,
+            y0,
+            xDomainCorrectionFactor,
+            focusSeriesName,
+            placedLabels,
+            labelLines,
+            placedItems,
+            hoveredEntityName,
+        } = this
+        const selectionSet = this.selectionArray.selectedSet
+        const targetTime = this.manager.endTime
+        const timeColumn = this.inputTable.timeColumn
+        const yAxisColumn = this.formatColumn
+        const xAxisColumn = this.xColumn
+        const labelYOffset = 0
+        const hasSelection = selectionSet.size > 0
+        let noDataAreaElement = undefined
+        let noDataLabel = undefined
+        let pattern: JSX.Element | undefined = undefined
+        const noDataHeight = Bounds.forText("no data").height + 10 //  dualAxis.verticalAxis.rangeSize
+
+        const firstNanValue = placedItems.findIndex((item) => !item.bars.length)
+        const anyNonNanAfterFirstNan =
+            firstNanValue >= 0
+                ? drop(placedItems, firstNanValue).some(
+                      (item) => item.bars.length !== 0
+                  )
+                : false
+
+        if (anyNonNanAfterFirstNan)
+            console.error("Found Non-NAN values after NAN value!")
+
+        if (firstNanValue !== -1) {
+            const firstNanValueItem = placedItems[firstNanValue]
+            const lastItem = last(placedItems)!
+            const noDataRangeStartX =
+                firstNanValueItem.xPosition + dualAxis.horizontalAxis.place(x0)
+            const noDataRangeEndX =
+                lastItem?.xPosition +
+                dualAxis.horizontalAxis.place(lastItem.xPoint.value)
+            const yStart = dualAxis.verticalAxis.place(y0)
+
+            noDataLabel = (
+                <text
+                    key={`noDataArea-label`}
+                    x={
+                        noDataRangeStartX +
+                        (noDataRangeEndX - noDataRangeStartX) / 2
+                    }
+                    y={yStart - noDataHeight / 2}
+                    width={noDataRangeEndX - noDataRangeStartX}
+                    height={noDataHeight}
+                    fontWeight={300}
+                    fill="#000"
+                    opacity={1}
+                    fontSize="1em"
+                    textAnchor="middle"
+                    dominantBaseline="middle"
+                >
+                    no data
```

Comment:
Nice, I like this new No data label way better!
You could add a `style={{pointerEvents: "none"}}`, this way the tooltip will show up even if the user hovers over the text itself.
Also, it would be great if the tooltip could also show "No data" instead of it just not showing anything regarding the y value at all.

Reply:
Nice tipp, thanks!